File: Q33.cpp
//Q33.cpp
class Solution {
public:
    int search(vector<int>& a, int t) {
        int l=0,r=a.size()-1;
        while(l<=r){
            int m=(l+r)>>1;
            if(a[m]==t) return m;
            if(a[l]<=a[m]){
                if(a[l]<=t && t<a[m]) r=m-1;
                else l=m+1;
            } else {
                if(a[m]<t && t<=a[r]) l=m+1;
                else r=m-1;
            }
        }
        return -1;
    }
};

File: Q33.py
#Q33.py
class Solution:
    def search(self, a, t):
        l=0; r=len(a)-1
        while l<=r:
            m=(l+r)//2
            if a[m]==t: return m
            if a[l]<=a[m]:
                if a[l]<=t<a[m]: r=m-1
                else: l=m+1
            else:
                if a[m]<t<=a[r]: l=m+1
                else: r=m-1
        return -1
